src/ldp/permissions/BasePermissionsExtractor.ts

```diff
@@ -0,0 +1,26 @@
+import { Operation } from '../operations/Operation';
+import { PermissionSet } from './PermissionSet';
+import { PermissionsExtractor } from './PermissionsExtractor';
+import { UnsupportedHttpError } from '../../util/errors/UnsupportedHttpError';
+
+/**
+ * Generates permissions for the base set of methods that always require the same permissions.
+ * Specifically: GET, HEAD, POST, PUT and DELETE.
+ */
+export class BasePermissionsExtractor extends PermissionsExtractor {
+  public async canHandle(input: Operation): Promise<void> {
+    if (![ 'HEAD', 'GET', 'POST', 'PUT', 'DELETE' ].includes(input.method)) {
+      throw new UnsupportedHttpError('Only HEAD, GET, POST, PUT and DELETE are supported.');
```

Comment:
Ridiculous nit, but:

```suggestion
    if (!/^(?:HEAD|GET|POST|PUT|DELETE)$/.test(input.method)) {
      throw new UnsupportedHttpError(`Unsupported method: ${input.method}`);
```

to avoid a) creating and looping over an array b) repeating the info.